### PR TITLE
update clusterset contoller to ignore non-legacy clusterset

### PR DESCRIPTION
**Pull request by elgnay**
Related task: https://github.com/stolostron/backlog/issues/20209
Signed-off-by: Yang Le <yangle@redhat.com>

### CODE REVIEW

File: pkg/hub/managedclusterset/controller.go [hunk 1]
```diff
@@ -135,6 +135,12 @@ func (c *managedClusterSetController) sync(ctx context.Context, syncCtx factory.
 		return nil
 	}
 
+	// only hand the legacy clusterset
```
Comment (qiujian16) [inline pkg/hub/managedclusterset/controller.go]: handle

File: pkg/hub/managedclusterset/controller.go [hunk 2]
```diff
@@ -135,6 +135,12 @@ func (c *managedClusterSetController) sync(ctx context.Context, syncCtx factory.
...
+	selectorType := clusterSet.Spec.ClusterSelector.SelectorType
+	if len(selectorType) > 0 && selectorType != clusterv1beta1.LegacyClusterSetLabel {
```
Comment (qiujian16) [inline pkg/hub/managedclusterset/controller.go]: This is still a bit strange to me, I think we should check the type in syncClusterSet, or we should at least add a comment indicating if more type is added in the future, this code should be removed.

File: pkg/hub/managedclusterset/controller.go (hunk 1 above)
Comment (elgnay) [reply pkg/hub/managedclusterset/controller.go]: fixed

File: pkg/hub/managedclusterset/controller.go (hunk 2 above)
Comment (elgnay) [reply pkg/hub/managedclusterset/controller.go]: Moved it into `syncClusterSet` as suggested.

**Review by qiujian16 — COMMENTED** [review]
/approve
/lgtm

**Pull request merged**